Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/shortcuts/AppShortcutsHelper.kt

```diff
@@ -1,64 +1,96 @@
 package de.rki.coronawarnapp.util.shortcuts
 
+import android.Manifest
 import android.content.Context
 import android.content.Intent
+import androidx.core.content.PermissionChecker
+import androidx.core.content.PermissionChecker.PERMISSION_GRANTED
 import androidx.core.content.pm.ShortcutInfoCompat
 import androidx.core.content.pm.ShortcutManagerCompat
 import androidx.core.graphics.drawable.IconCompat
 import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.coronatest.CoronaTestRepository
+import de.rki.coronawarnapp.storage.OnboardingSettings
 import de.rki.coronawarnapp.ui.launcher.LauncherActivity
 import de.rki.coronawarnapp.util.AppShortcuts
 import de.rki.coronawarnapp.util.di.AppContext
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.withContext
+import timber.log.Timber
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class AppShortcutsHelper @Inject constructor(@AppContext private val context: Context) {
-
+class AppShortcutsHelper @Inject constructor(
+    @AppContext private val context: Context,
+    private val onboardingSettings: OnboardingSettings,
+    private val coronaTestRepository: CoronaTestRepository
+) {
     suspend fun restoreAppShortcut() = withContext(Dispatchers.IO) {
+        // No shortcuts if test result is positive
+        if (coronaTestRepository.allCoronaTests.first().any { it.isPositive }) {
```

Comment:
is it specified somewhere to disable shortcuts for positive tested users ? I think this is irrelevant to shortcuts

Reply:
As I mentioned above, some similar check is performed in the iOS app, but I don't really know what the reason is: https://github.com/corona-warn-app/cwa-app-ios/blob/release/2.19.x/src/xcode/ENA/ENA/Source/AppDelegate%20%26%20Globals/AppDelegate%2BQuickActions.swift#L40-L44

Reply:
This check in iOS to prevent loosing the exposure keys while submitting the keys during sharing with others. However I don't think we need it in Android if I'm not mistaken. Anyway story has been prioritised and we will have a look into soon, I ask kindly for your patience

Reply:
Alright, thanks for clarification. I removed the check.